aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -942,6 +943,38 @@ impl AptosVM {
         simulation_vm.simulate_signed_transaction(&state_view.as_move_resolver(), txn, &log_context)
     }
 
+    pub fn execute_view_function(
+        state_view: &impl StateView,
+        module_id: ModuleId,
+        func_name: Identifier,
+        arguments: Vec<Vec<u8>>,
+        type_args: Vec<TypeTag>,
+        gas_budget: u64,
+    ) -> Result<Vec<Vec<u8>>> {
+        let vm = AptosVM::new(state_view);
+        let log_context = AdapterLogSchema::new(state_view.id(), 0);
+        let mut gas_meter = AptosGasMeter::new(
+            vm.0.get_gas_feature_version(),
+            vm.0.get_gas_parameters(&log_context)?.clone(),
+            vm.0.get_storage_gas_parameters(&log_context)?.clone(),
+            gas_budget,
+        );
+        Ok(vm
+            .new_session(&state_view.as_move_resolver(), SessionId::Void)
+            .execute_function_bypass_visibility(
+                &module_id,
+                func_name.as_ident_str(),
+                type_args,
+                arguments,
+                &mut gas_meter,
+            )
+            .map_err(|err| anyhow!("Failed to execute function: {:?}", err))?
+            .return_values
+            .into_iter()
+            .map(|(bytes, _ty)| bytes)
+            .collect::<Vec<_>>())
+    }
```

Comment:
So, I noticed this in the final output:

Do you expect there to be a way to return multiple values from a Move function?  Wouldn't this be handled by returning a vector type.

I think this should be Result<Vec<u8>> and not Result<Vec<Vec<u8>> right?

Reply:
This is exactly why we need `Result<Vec<Vec>`. It's a vector of return values, each of them is a `Vec<u8>`